src/components/molecules/popover.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+
+import { FloatingFocusManager, FloatingPortal, useMergeRefs } from "@floating-ui/react";
+
+import { PopoverContext, usePopoverContext } from "@contexts";
+import { usePopover } from "@src/hooks";
+import { PopoverOptions, PopoverTriggerProps } from "@src/interfaces/components";
+
+export const Popover = ({
+	children,
+	modal = false,
+	...restOptions
+}: {
+	children: React.ReactNode;
+} & PopoverOptions) => {
+	const popover = usePopover({ modal, ...restOptions });
+
+	return <PopoverContext.Provider value={popover}>{children}</PopoverContext.Provider>;
+};
+
+export const PopoverTrigger = React.forwardRef<HTMLElement, React.HTMLProps<HTMLElement> & PopoverTriggerProps>(
+	function PopoverTrigger({ children, ...props }, propRef) {
+		const context = usePopoverContext();
+		const childrenRef = (children as any).ref;
```

Comment:
is it possible not to use  type `any`?

Reply:
No, because children can be: number, string, ReactNode and on...
But I refactored it to look a little better.